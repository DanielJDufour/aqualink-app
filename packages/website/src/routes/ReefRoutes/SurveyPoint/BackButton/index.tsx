import React from "react";
import {
  Button,
  Container,
  Grid,
  Typography,
  withStyles,
  WithStyles,
  createStyles,
} from "@material-ui/core";
import { ArrowBack } from "@material-ui/icons";
import { Link } from "react-router-dom";

const BackButton = ({ reefId, classes }: BackButtonProps) => {
  return (
    <Container>
      <Grid container className={classes.backButtonWrapper}>
        <Button
          color="primary"
          startIcon={<ArrowBack />}
          component={Link}
          to={`/reefs/${reefId}`}
        >
          <Typography className={classes.backButtonText}>
            Back to Site
          </Typography>
        </Button>
      </Grid>
    </Container>
  );
};

const styles = () =>
  createStyles({
    backButtonWrapper: {
      margin: "48px 0 16px 0",
    },

    backButtonText: {
      textTransform: "none",
    },
  });

interface BackButtonIncomingProps {
  reefId: string;
}

type BackButtonProps = BackButtonIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(BackButton);
