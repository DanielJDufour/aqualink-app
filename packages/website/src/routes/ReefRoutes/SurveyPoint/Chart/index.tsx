import React, { useEffect, useState } from "react";
import {
  Container,
  Grid,
  withStyles,
  WithStyles,
  createStyles,
  Theme,
} from "@material-ui/core";
import moment from "moment";
import { useDispatch, useSelector } from "react-redux";

import Chart from "./Chart";
import TempAnalysis from "./TempAnalysis";
import {
  reefHoboDataRequest,
  reefSpotterDataRequest,
  reefSpotterDataSelector,
} from "../../../../store/Reefs/selectedReefSlice";
import { Metrics, Reef } from "../../../../store/Reefs/types";
import { setTimeZone, subtractFromDate } from "../../../../helpers/dates";

const ChartWithCard = ({ reef, pointId, classes }: ChartWithCardProps) => {
  const dispatch = useDispatch();
  const spotterData = useSelector(reefSpotterDataSelector);
  const hasSpotterData = Boolean(reef.liveData.surfaceTemperature);
  const [pickerEndDate, setPickerEndDate] = useState<string>(
    new Date(moment().format("MM/DD/YYYY")).toISOString()
  );
  const [pickerStartDate, setPickerStartDate] = useState<string>(
    subtractFromDate(
      new Date(moment().format("MM/DD/YYYY")).toISOString(),
      "week"
    )
  );

  // Get spotter data
  useEffect(() => {
    const reefLocalStartDate = setTimeZone(
      new Date(pickerStartDate),
      reef.timezone
    ) as string;

    const reefLocalEndDate = setTimeZone(
      new Date(pickerEndDate),
      reef.timezone
    ) as string;

    dispatch(
      reefHoboDataRequest({
        reefId: `${reef.id}`,
        pointId,
        start: reefLocalStartDate,
        end: reefLocalEndDate,
        metrics: [Metrics.bottomTemperature],
      })
    );

    if (hasSpotterData) {
      dispatch(
        reefSpotterDataRequest({
          id: `${reef.id}`,
          startDate: reefLocalStartDate,
          endDate: reefLocalEndDate,
        })
      );
    }
  }, [
    dispatch,
    hasSpotterData,
    pickerEndDate,
    pickerStartDate,
    pointId,
    reef.id,
    reef.timezone,
  ]);

  return (
    <Container>
      {hasSpotterData && (
        <Grid className={classes.chartWrapper} container item spacing={2}>
          <Grid item xs={12} md={9}>
            <Chart
              reef={reef}
              spotterData={spotterData}
              pickerStartDate={pickerStartDate}
              pickerEndDate={pickerEndDate}
              onStartDateChange={(date) =>
                setPickerStartDate(
                  new Date(moment(date).format("MM/DD/YYYY")).toISOString()
                )
              }
              onEndDateChange={(date) =>
                setPickerEndDate(
                  new Date(moment(date).format("MM/DD/YYYY")).toISOString()
                )
              }
            />
          </Grid>
          <Grid item xs={12} md={3}>
            <Grid container justify="center">
              <Grid item xs={11} sm={5} md={11} lg={10}>
                <TempAnalysis
                  startDate={pickerStartDate}
                  endDate={pickerEndDate}
                  depth={reef.depth}
                  spotterData={spotterData}
                />
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      )}
    </Container>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    chartWrapper: {
      margin: "80px 0 20px 0",
      [theme.breakpoints.down("xs")]: {
        margin: "40px 0 10px 0",
      },
    },
  });

interface ChartWithCardIncomingProps {
  reef: Reef;
  pointId: string;
}

type ChartWithCardProps = ChartWithCardIncomingProps &
  WithStyles<typeof styles>;

export default withStyles(styles)(ChartWithCard);
