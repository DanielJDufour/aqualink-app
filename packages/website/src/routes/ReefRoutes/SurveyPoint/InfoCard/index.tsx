import React from "react";
import {
  Container,
  Card,
  withStyles,
  WithStyles,
  createStyles,
  Grid,
} from "@material-ui/core";

import Info from "./Info";
import Map from "./Map";
import { Reef } from "../../../../store/Reefs/types";

const InfoCard = ({ reef, pointId, classes }: InfoCardProps) => {
  return (
    <Container>
      <Grid className={classes.cardWrapper} container justify="center">
        <Grid item xs={12} sm={12}>
          <Card elevation={3}>
            <Grid container justify="space-between">
              <Info reef={reef} pointId={pointId} />
              <Map reef={reef} />
            </Grid>
          </Card>
        </Grid>
      </Grid>
    </Container>
  );
};

const styles = () =>
  createStyles({
    cardWrapper: {
      marginBottom: 100,
    },
  });

interface InfoCardIncomingProps {
  reef: Reef;
  pointId: number;
}

type InfoCardProps = InfoCardIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(InfoCard);
