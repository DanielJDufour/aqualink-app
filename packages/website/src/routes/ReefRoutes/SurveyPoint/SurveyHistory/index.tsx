import React from "react";
import {
  Container,
  Box,
  withStyles,
  WithStyles,
  createStyles,
  Typography,
  Grid,
  Theme,
} from "@material-ui/core";
import { useSelector } from "react-redux";

import TimeLine from "../../../../common/SiteDetails/Surveys/Timeline";
import { Reef } from "../../../../store/Reefs/types";
import { userInfoSelector } from "../../../../store/User/userSlice";
import { isAdmin } from "../../../../helpers/user";

const SurveyHistory = ({ reef, pointId, classes }: SurveyHistoryProps) => {
  const user = useSelector(userInfoSelector);
  const { name: pointName } =
    reef?.surveyPoints.filter((point) => point.id === pointId)[0] || {};

  return (
    <Box className={classes.timelineWrapper}>
      <Container>
        <Grid container justify="center">
          <Box className={classes.title}>
            <Typography variant="h4">{pointName} Survey History</Typography>
          </Box>
        </Grid>
        <TimeLine
          isAdmin={isAdmin(user, reef.id)}
          addNewButton={false}
          observation="any"
          pointName={pointName}
          pointId={pointId}
          reefId={reef.id}
          timeZone={reef.timezone}
        />
      </Container>
    </Box>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    timelineWrapper: {
      backgroundColor: "rgb(245, 246, 246)",
    },
    title: {
      marginTop: 100,
      [theme.breakpoints.down("xs")]: {
        marginTop: 50,
      },
    },
  });

interface SurveyHistoryIncomingProps {
  reef: Reef;
  pointId: number;
}

type SurveyHistoryProps = SurveyHistoryIncomingProps &
  WithStyles<typeof styles>;

export default withStyles(styles)(SurveyHistory);
