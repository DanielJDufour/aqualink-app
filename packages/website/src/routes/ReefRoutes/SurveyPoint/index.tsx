import React from "react";
import { withStyles, WithStyles, createStyles } from "@material-ui/core";

const SurveyPoint = ({ classes }: SurveyPointProps) => {
  return (
    <div className={classes.root}>
      <span>Survey Point</span>
    </div>
  );
};

const styles = () =>
  createStyles({
    root: {},
  });

interface SurveyPointIncomingProps {}

type SurveyPointProps = SurveyPointIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(SurveyPoint);
