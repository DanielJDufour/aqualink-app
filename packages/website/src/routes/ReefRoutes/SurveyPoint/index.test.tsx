import React from "react";
import { render } from "@testing-library/react";

import SurveyPoint from ".";

test("renders as expected", () => {
  const { container } = render(<SurveyPoint />);
  expect(container).toMatchSnapshot();
});
