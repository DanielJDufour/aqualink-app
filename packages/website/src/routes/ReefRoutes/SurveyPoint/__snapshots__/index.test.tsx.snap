// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="SurveyPoint-root-1"
  >
    <span>
      Survey Point
    </span>
  </div>
</div>
`;
